### PR TITLE
Add note about missing mixed type support to docs

### LINKED ISSUE #1981: Type inference not consistent with pandas
The behavior of creating a dataframe with mixed types is different in Koalas from that in Pandas.  Pandas correctly infers that the type of a column with mixed types should be `object`.  

Example:

```
>>> a = pd.DataFrame({'col1': [1, 'a'], 'col2': [3, 4]})
>>> a.dtypes
col1    object
col2     int64
dtype: object
>>> b = ks.DataFrame({'col1': [1, 'a'], 'col2': [3, 4]}) # throws error
```

This behavior isn't clear from the documentation as https://koalas.readthedocs.io/en/latest/user_guide/types.html implies that the Pandas object type is fully supported.

There are many real-world unclean datasets where types are mixed. It would be nice if Koalas could support them in a Pandas like way.

### PR DESCRIPTION
**Pull request by itholic**
Added note about missing support for mixed type to documents.

Screen Shot 2021-01-05 at 9 36 39 AM

Resolves #1981

### CODE REVIEW

File: docs/source/user_guide/types.rst [hunk 1]
```diff
@@ -6,6 +6,8 @@ Type Support In Koalas
 
 In this chapter, we will briefly show you how data types change when converting Koalas DataFrame from/to PySpark DataFrame or pandas DataFrame.
 
+.. note::
+    Koalas doesn't support mixed type because PySpark doesn't support it for now.
```
Comment (HyukjinKwon) [inline docs/source/user_guide/types.rst:10]: I can't tell what this means from reading this sentence.

Comment (itholic) [reply docs/source/user_guide/types.rst:10]: I'll improve the explanation with some code examples.

File: docs/source/user_guide/types.rst [hunk 2]
```diff
@@ -6,7 +6,6 @@ Type Support In Koalas
 
 In this chapter, we will briefly show you how data types change when converting Koalas DataFrame from/to PySpark DataFrame or pandas DataFrame.
 
-
```
Comment (HyukjinKwon) [inline docs/source/user_guide/types.rst:9]: Can you remove the unrelated change?

File: docs/source/user_guide/types.rst [hunk 3]
```diff
@@ -215,3 +214,14 @@ You can also check the underlying PySpark data type of `Series` or schema of `Da
      |-- d: double (nullable = false)
      |-- s: string (nullable = false)
      |-- b: boolean (nullable = false)
+
+.. note::
+    Koalas currently does not support multiple types of data in single column.
+    This is because Koalas uses Spark DataFrame internally and Spark DataFrame does not support multiple types of data in single column.
```
Comment (HyukjinKwon) [inline docs/source/user_guide/types.rst:220]: Use consistent wording. Looks like this page used PySpark DataFrame instead of Spark DataFrame.

Comment (HyukjinKwon) [inline docs/source/user_guide/types.rst:220]: I think you can just remove "This is because Koalas uses Spark DataFrame internally and Spark DataFrame does not support multiple types of data in single column.".

File: docs/source/user_guide/types.rst [hunk 4]
```diff
@@ -215,3 +214,14 @@ You can also check the underlying PySpark data type of `Series` or schema of `Da
      |-- d: double (nullable = false)
      |-- s: string (nullable = false)
      |-- b: boolean (nullable = false)
+
+.. note::
```
Comment (HyukjinKwon) [inline docs/source/user_guide/types.rst:218]: Can you add a newline after this? See https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-note

File: docs/source/user_guide/types.rst (hunk 3 above)
Comment (itholic) [reply docs/source/user_guide/types.rst:220]: Seems more clearer. Thanks!

**Comment by itholic** [comment]
Thanks, @HyukjinKwon :)

File: docs/source/user_guide/types.rst [hunk 5]
```diff
@@ -215,3 +215,14 @@ You can also check the underlying PySpark data type of `Series` or schema of `Da
...
+
+    Koalas currently does not support multiple types of data in single column.
+
+.. code-block:: python
```
Comment (HyukjinKwon) [inline docs/source/user_guide/types.rst:223]: Can you add this example inside the note?

Comment (itholic) [reply docs/source/user_guide/types.rst]: Addressed. Thanks!

**Comment by HyukjinKwon** [comment]
Can you update PR description too?

**Comment by itholic** [comment]
Updated!

**Review by HyukjinKwon — APPROVED** [review]
